### DISCUSSION TITLE
freezing checkbox state

### DISCUSSION THREAD

**Discussion by vishnuathreya28** [mCaptcha/mCaptcha#159] [Q&A]
is it possible to freeze the state of the checkbox once checked to ensure that the user can't check and uncheck multiple times unless the page is refreshed?
if yes what files should be edited to make this change?

thanks in advance

**Comment by realaravinth** [comment]
Hello Vishnu o/

Captcha solutions expire after [cooldown period](https://mcaptcha.org/docs/webmasters/terminology#cooldown-period). I originally left it editable, so that I can uncheck it from code after the solution expires. This functionally doesn't exist yet, though.

> if yes what files should be edited to make this change?

IMO, this is an anti-goal for the project right now (open to change if others chime in). However, if you are willing to maintain a fork of the library, then you can lock the checkbox by setting it to `readonly` in [this hook](https://github.com/mCaptcha/glue/blob/d66d22f6c869c8d3e49efabff7078f15a493e9b8/packages/vanilla/src/widget.ts#L72) or the equivalent glue library for your framework.

**Reply by vishnuathreya28** [reply]
Thank you for reaching out. The challenge I'm currently navigating involves integrating this functionality into a JSP environment, which unfortunately lacks an equivalent library akin to Glue. To proceed, I'm seeking guidance on which specific HTML or JS file requires modification to implement the desired functionality. Any insights or suggestions would be greatly appreciated.

**Reply by realaravinth** [reply]
I'm sorry, I don't understand. JSP is a backend tech, no? The checkbox is controlled by [mCaptcha/glue](https://github.com/mcaptcha/glue), which is a set of frontend libraries to integrate mCaptcha. It doesn't support freezing the checkbox after a successful CAPTCHA solve. To get that feature, you'll have to fork mCaptcha/glue, and [edit this hook](https://github.com/mCaptcha/glue/blob/d66d22f6c869c8d3e49efabff7078f15a493e9b8/packages/vanilla/src/widget.ts#L72).
